src/engine/Bind.cpp

```diff
@@ -81,104 +81,138 @@ std::vector<QueryExecutionTree*> Bind::getChildren() {
+  std::ranges::copy(idTable.begin() + subrange.first,
+                    idTable.begin() + subrange.second, result.begin());
+  return result;
 }
 
 // _____________________________________________________________________________
-template <size_t IN_WIDTH, size_t OUT_WIDTH>
-void Bind::computeExpressionBind(
-    IdTable* outputIdTable, LocalVocab* outputLocalVocab,
-    const Result& inputResultTable,
-    sparqlExpression::SparqlExpression* expression) const {
+ProtoResult Bind::computeResult(bool requestLaziness) {
+  LOG(DEBUG) << "Get input to BIND operation..." << std::endl;
+  std::shared_ptr<const Result> subRes = _subtree->getResult(requestLaziness);
+  LOG(DEBUG) << "Got input to Bind operation." << std::endl;
+
+  auto applyBind = [this, subRes](
+                       auto&& idTable, LocalVocab* localVocab,
+                       std::optional<std::pair<size_t, size_t>> subrange =
+                           std::nullopt) {
+    return computeExpressionBind(localVocab, AD_FWD(idTable),
+                                 subRes->localVocab(),
+                                 _bind._expression.getPimpl(), subrange);
+  };
+
+  if (subRes->isFullyMaterialized()) {
+    if (requestLaziness && subRes->idTable().size() > CHUNK_SIZE) {
+      auto localVocab =
+          std::make_shared<LocalVocab>(subRes->getCopyOfLocalVocab());
+      auto generator = [](std::shared_ptr<LocalVocab> vocab, auto applyBind,
+                          std::shared_ptr<const Result> result)
+          -> cppcoro::generator<IdTable> {
+        size_t size = result->idTable().size();
+        for (size_t offset = 0; offset < size; offset += CHUNK_SIZE) {
+          co_yield applyBind(result->idTable(), vocab.get(),
+                             {{offset, std::min(size, offset + CHUNK_SIZE)}});
+        }
+      }(localVocab, std::move(applyBind), std::move(subRes));
+      return {std::move(generator), resultSortedOn(), std::move(localVocab)};
+    }
+    // Make a deep copy of the local vocab from `subRes` and then add to it (in
+    // case BIND adds a new word or words).
+    //
+    // TODO: In most BIND operations, nothing is added to the local vocabulary,
+    // so it would be more efficient to first share the pointer here (like with
+    // `shareLocalVocabFrom`) and only copy it when a new word is about to be
+    // added. Same for GROUP BY.
+    LocalVocab localVocab = subRes->getCopyOfLocalVocab();
+    IdTable result = applyBind(subRes->idTable(), &localVocab);
+    LOG(DEBUG) << "BIND result computation done." << std::endl;
+    return {std::move(result), resultSortedOn(), std::move(localVocab)};
+  }
+  auto localVocab = std::make_shared<LocalVocab>();
+  auto generator =
+      [](std::shared_ptr<LocalVocab> vocab, auto applyBind,
+         std::shared_ptr<const Result> result) -> cppcoro::generator<IdTable> {
+    for (IdTable& idTable : result->idTables()) {
+      co_yield applyBind(idTable, vocab.get());
```

Comment:
```suggestion
      co_yield applyBind(std::move(idTable), vocab.get());
```
s.t. you can `AD_FWD`inside `computeExpressionBind` without triggering a copy.